src/test/java/ro/developmentfactory/thedrones/service/DronePositionServiceImplTest.java

```diff
@@ -0,0 +1,267 @@
+    }
+
+
+    @Test
+    @DisplayName("Move Forward: Multiple valid moves should update position correctly")
+    void moveForward_MultipleMoves_ShouldUpdatePositionCorrectly() {
+        // Given
+        droneStatus.setCurrentPositionX(0);
+        droneStatus.setCurrentPositionY(0);
+        droneStatus.setFacingDirection(Direction.E);
+        when(droneRepository.findById(any(UUID.class))).thenReturn(Optional.of(drone));
+
+        // When
+        dronePositionService.moveForward(drone.getIdDrone()); // Move to (1, 0)
+        dronePositionService.moveForward(drone.getIdDrone()); // Move to (2, 0)
+
+        // Then
+        assertEquals(2, droneStatus.getCurrentPositionX());
+        assertEquals(0, droneStatus.getCurrentPositionY());
+        assertEquals(Direction.E, droneStatus.getFacingDirection());
+        assertEquals(2, drone.getCountMove());
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "N,E",
+            "E,S",
+            "S,W",
+            "W,N"
+    })
+    @DisplayName("Turning Direction: Valid turn should update direction correctly")
+    void turningDirection_ValidTurn_ShouldUpdateDirection(Direction initialDirection, Direction expectedDirection) {
+        // Given
+        droneStatus.setFacingDirection(initialDirection);
+        when(droneRepository.findById(any(UUID.class))).thenReturn(Optional.of(drone));
+
+        // When
+        DroneStatusResponse response = dronePositionService.turn(drone.getIdDrone(), TurnDirection.RIGHT);
+
+        // Then
+        assertEquals(expectedDirection, response.getFacingDirection());
+        verify(droneStatusRepository).save(droneStatus);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "0,0,E,1,0",
+            "0,0,N,0,1",
+            "1,1,W,0,1",
+            "1,1,S,1,0"
+    })
+    @DisplayName("Move Forward: Valid moves should update position correctly")
+    void moveForward_ValidMoves_ShouldUpdatePositionCorrectly(int startX, int startY, Direction direction, int expectedX, int expectedY) {
+        // Given
+        droneStatus.setCurrentPositionX(startX);
+        droneStatus.setCurrentPositionY(startY);
+        droneStatus.setFacingDirection(direction);
+        when(droneRepository.findById(any(UUID.class))).thenReturn(Optional.of(drone));
+
+        // When
+        DroneStatusResponse response = dronePositionService.moveForward(drone.getIdDrone());
+
+        // Then
+        assertEquals(expectedX, response.getCurrentPositionX());
+        assertEquals(expectedY, response.getCurrentPositionY());
+        assertEquals(direction, response.getFacingDirection());
+        verify(droneStatusRepository).save(droneStatus);
+    }
+
+    @ParameterizedTest
+    @EnumSource(Direction.class)
+    @DisplayName("Turning Left: Valid turn should update direction correctly")
+    void turningLeft_ValidTurn_ShouldUpdateDirection(Direction initialDirection) {
```

Comment:
you don't need this test because `turningDirection_ValidRotation_ShouldUpdateDirectionCorrectly` already covers it